src/components/NavBar/NavBar.tsx

```diff
@@ -1,15 +1,27 @@
 'use client';
 
-import { Button, Flex, Image, View } from '@aws-amplify/ui-react';
-import Link from 'next/link'; // Import Link from next/link
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { Button, Flex, Image, View, Menu, MenuButton, MenuItem } from '@aws-amplify/ui-react';
+import Link from 'next/link';
 import { useTranslation } from 'react-i18next';
 import styles from './navbar.module.css';
 
 const NavBar = () => {
-  const [currentLangCode, setCurrentLangCode] = useState(
-    window.localStorage.i18nextLng || 'en'
-  );
+
+  const [currentLangCode, setCurrentLangCode] = useState('en');
+  const [isMobile, setIsMobile] = useState(false);
+
+
+  useEffect(() => {
```

Comment:
I think you'll want to use the useBreakpointValue hook for this instead
https://ui.docs.amplify.aws/react/theming/responsive#breakpoints

Reply:
The documentation on amplify isn't super great but I'm assuming it works similarly to the material ui media query hook https://mui.com/material-ui/react-use-media-query/